odoo/addons/base/tests/test_base.py

```diff
@@ -501,6 +501,107 @@ def test_60_read_group(self):
         self.assertEqual([2, 4], [g['title_count'] for g in groups_data], 'Incorrect number of results')
         self.assertEqual([-1, 10], [g['color'] for g in groups_data], 'Incorrect aggregation of int column')
 
+    def test_61_private_read_group(self):
+        """
+        the _read_group should behave exactly like read_group (public method) except for sorting the one2many on ID
+        instead of name, so avoiding the join on the "to many" table to get the name
+        """
+        title_sir = self.env['res.partner.title'].create({'name': 'Sir...'})
+        title_lady = self.env['res.partner.title'].create({'name': 'Lady...'})
+        test_users = [
+            {'name': 'Alice', 'login': 'alice', 'color': 1, 'function': 'Friend', 'date': '2015-03-28', 'title': title_lady.id},
+            {'name': 'Alice', 'login': 'alice2', 'color': 0, 'function': 'Friend', 'date': '2015-01-28', 'title': title_lady.id},
+            {'name': 'Bob', 'login': 'bob', 'color': 2, 'function': 'Friend', 'date': '2015-03-02', 'title': title_sir.id},
+            {'name': 'Eve', 'login': 'eve', 'color': 3, 'function': 'Eavesdropper', 'date': '2015-03-20', 'title': title_lady.id},
+            {'name': 'Nab', 'login': 'nab', 'color': -3, 'function': '5$ Wrench', 'date': '2014-09-10', 'title': title_sir.id},
+            {'name': 'Nab', 'login': 'nab-she', 'color': 6, 'function': '5$ Wrench', 'date': '2014-01-02', 'title': title_lady.id},
+        ]
+        res_users = self.env['res.users']
+        user_ids = [res_users.create(vals).id for vals in test_users]
+        domain = [('id', 'in', user_ids)]
```

Comment:
You should batch creation as:
```suggestion
        users = self.env['res.users'].create(test_users)
        domain = [('id', 'in', users.ids)]
```